cql3/statements/create_keyspace_statement.cc

```diff
@@ -146,20 +147,68 @@ future<> cql3::statements::create_keyspace_statement::grant_permissions_to_creat
     });
 }
 
+using strategy_class_registry = class_registry<
+    locator::abstract_replication_strategy,
+    const sstring&,
+    const locator::shared_token_metadata&,
+    locator::snitch_ptr&,
+    const std::map<sstring, sstring>&>;
+
+// Check for replication strategy choices which are restricted by the
+// configuration. This check can throw a configuration_exception immediately
+// if the strategy is forbidden by the configuration, or return a warning
+// string if the restriction was set to "warn".
+// This function is only supposed to check for replication strategies
+// restricted by the configuration. Checks for other types of strategy
+// errors (such as unknown replication strategy name or unknown options
+// to a known replication strategy) are done elsewhere.
+std::optional<sstring> check_restricted_replication_strategy(
+    service::storage_proxy& proxy,
+    const sstring& keyspace,
+    const ks_prop_defs& attrs)
+{
+    if (!attrs.get_replication_strategy_class()) {
+        return std::nullopt;
+    }
+    sstring replication_strategy = strategy_class_registry::to_qualified_class_name(
+        *attrs.get_replication_strategy_class());
+    // SimpleStrategy is not recommended in any setup which already has - or
+    // may have in the future - multiple racks or DCs. So depending on how
+    // protective we are configured, let's prevent it or allow with a warning:
+    if (replication_strategy == "org.apache.cassandra.locator.SimpleStrategy") {
+        auto restricted = proxy.local_db().get_config().restrict_replication_simplestrategy();
+        if (restricted == db::tri_mode_restriction_t::TRUE) {
```

Comment:
I'm not an expert in our enum wrappers, but can't they be used in a `switch ()` statement? It looks like a natural candidate instead of a chain of `if` statements and comes with a compiler check for properly implementing cases for all possible values; if it can't be easily done, feel free to ignore this comment

Reply:
I tried (I also wanted to use switch...), and it didn't work because our wrapper is not a db::tri_mode_restriction_t - it can only be compared to one. I can try again, I'll check if I can add some sort of implicit converter that will make the switch work.
Note that even the "==" didn't originally work - I sent a previous patch (which was already committed) allowing operator== between the wrapper type and its content type, so this is why the "==" worked.

Reply:
It does work after adding the appropriate implicit converter.